src/module/actor/DamageApplicationFlow.ts

```diff
@@ -0,0 +1,112 @@
+import {DamageAndSoakPerActor, DamageApplicationDialog} from "../apps/dialogs/DamageApplicationDialog";
+import { SR5 } from "../config";
+import {SR5Actor} from "./SR5Actor";
+import DamageData = Shadowrun.DamageData;
+import CharacterActorData = Shadowrun.CharacterActorData;
+
+export class DamageApplicationFlow {
+
+    /**
+     * Runs the flow to apply damage to multiple actors with user interaction
+     * This will also take care of changing the damage type if necessary
+     * @param actors The actors that are affected
+     * @param damage The damage the actors will receive
+     */
+    async runApplyDamage(actors: SR5Actor[], damage : DamageData) {
+ 
+        console.log("Run apply damage");
+        const actorDamage : DamageAndSoakPerActor[] = 
+            actors.map(actor => this.getDamageAndSoakForActor(actor, damage));
+
+        // Show user the affected actors and the damage values
+        const damageApplicationDialog = await new DamageApplicationDialog(actorDamage, damage);
+        await damageApplicationDialog.select();
+
+        if (damageApplicationDialog.canceled) {
+            return;
+        }
+
+        if (damageApplicationDialog.selectedButton === 'damage') {
+            actorDamage.forEach(({actor, modified }) => {
+                this.applyDamageToActor(actor, modified);
+            });
+        }
+        else if (damageApplicationDialog.selectedButton === 'unmodifiedDamage') {
```

Comment:
The unmodifiedDamage button doesn't exist anymore. Is the different handling for damage and unmodifiedDamage still needed?

Reply:
It's also a bit confusing between what damage and modified mean, which is my fault. As I left it that way after removing said button and the functionality that needed both the button and an incoming vs modified damage.